### PR TITLE
General tweaks and enhancements

### PR DESCRIPTION
**Pull request by forgxyz**
# Description

_Please include a summary of changes and related issue (if any)._
Related to issue #58 - list of changes included there.

# Tests

- [ ] Please provide evidence of your successful `dbt run` / `dbt test` here
- [ ] Any comparison between `prod` and `dev` for any schema change

Running a full refresh on dev to check for any issues, will likely take overnight so results will be added later along with an incremental run and tests.

# Checklist
- [x] Follow [dbt style guide](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md)
- [x] Tag the person(s) responsible for reviewing proposed changes
- [x] Notes to deployment, if a `full-refresh` is needed for any table

Notable model changes:
- Updating the cluster_by to `cluster_by = ['ingested_at::DATE', 'block_timestamp::DATE']` because we use `ingested_at` in the incremental, but end-users will use `block_timestamp`
- Changing metrics models to be incremental

### CODE REVIEW

**Comment by forgxyz** [comment]
```
dbt run -m transactions+ --full-refresh
23:26:35  Found 12 models, 211 tests, 0 snapshots, 0 analyses, 605 macros, 0 operations, 0 seed files, 8 sources, 0 exposures, 0 metrics
23:26:36  Concurrency: 4 threads (target='default')
05:48:00  Finished running 9 incremental models in 22885.05s.
05:48:00  Completed with 1 error and 0 warnings:
05:48:00  Database Error in model receipts (models/core/receipts.sql)
05:48:00    000904 (42000): SQL compilation error: error line 39 at position 24
05:48:00    invalid identifier 'INGESTED_AT'
05:48:00    compiled SQL at target/run/near/models/core/receipts.sql
05:48:00  Done. PASS=8 WARN=0 ERROR=1 SKIP=0 TOTAL=9
```

Error in receipts was due to the change adding ingested_at to the cluster, but not ensuring ingested_at is in the model, but the rest were successful. Running transactions atm.

```
dbt test
...
06:38:31  Finished running 211 tests in 986.95s.
06:38:31
06:38:31  Completed with 1 error and 16 warnings:
06:38:31
06:38:31  Failure in test sequence_gaps_blocks_block_height (models/core/blocks.yml)
06:38:31    Got 49033 results, configured to fail if >25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/sequence_gaps_blocks_block_height.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_actions_events_addkey_allowance__NUMBER__None (models/core/actions_events_addkey.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/actions_events_addkey.yml/dbt_expectations_expect_column_8f50adaaf780e9e0a42b0bc88d3f9e15.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_block_challenges_result__ARRAY (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_b26f6cad7555e0bbe6539e3455a714b0.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_chunk_mask__ARRAY (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_f5de2e7a0293ed723a0f2cb113e5c66a.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_chunks__ARRAY (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_d727ac588a858a5dc0607c185d90ccaa.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_events__ARRAY (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_4aacf6552ce7e6fe93778fba407fcd1d.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_gas_price__NUMBER (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_6c3d2e8b80d20ed8288e4f2f42f13d0a.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_rent_paid__NUMBER (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_a59d45644327648d41be28e9e7fe4212.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_total_supply__NUMBER (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_0d811d8cb1289afb8900fa06703a1ac8.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_validator_proposals__ARRAY (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_da06b53cf5c51b32d24f8ef33b0d61e6.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_blocks_validator_reward__NUMBER (models/core/blocks.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/blocks.yml/dbt_expectations_expect_column_3e1d026a4643474b5a029c759753aebd.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_metrics_daily_gas_avg_gas_price__NUMBER (models/metrics/metrics_daily_gas.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/metrics/metrics_daily_gas.yml/dbt_expectations_expect_column_1805ea2dcc21d86bca56ece0878f4dd3.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_receipts_logs__ARRAY (models/core/receipts.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/receipts.yml/dbt_expectations_expect_column_4461347252ef2d2a125ed3bf890e0e99.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_receipts_proof__ARRAY (models/core/receipts.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/receipts.yml/dbt_expectations_expect_column_b8068db1b903cf306b28977476949ef6.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_receipts_receipt_outcome_id__ARRAY (models/core/receipts.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/receipts.yml/dbt_expectations_expect_column_67baaf912e6113673308d1337486535f.sql
06:38:31
06:38:31  Warning in test dbt_expectations_expect_column_values_to_be_in_type_list_transactions_tx_receipt__ARRAY (models/core/transactions.yml)
06:38:31    Got 1 result, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/transactions.yml/dbt_expectations_expect_column_2d7df3e7d6fa4f2eb982cae3c57f3571.sql
06:38:31
06:38:31  Warning in test tx_gaps_transactions_block_height__txn_hash__tx_count (models/core/transactions.yml)
06:38:31    Got 8 results, configured to warn if between 1 and 25000
06:38:31
06:38:31    compiled SQL at target/compiled/near/models/core/transactions.yml/tx_gaps_transactions_block_height__txn_hash__tx_count.sql
06:38:31
06:38:31  Done. PASS=194 WARN=16 ERROR=1 SKIP=0 TOTAL=211
```


Need to look at the trigger for these type expectation tests, for example the following only returns `ARRAY`, which is what we want to see but, as configured, the test is tripping a warning.
`select distinct typeof(validator_proposals) from near.dev.blocks`

**Comment by forgxyz** [comment]
No issues with receipts after column fix:

```
➜  near_dbt git:(hotfix-58) dbt run -m receipts --full-refresh
15:38:24  Running with dbt=1.1.0
15:38:25  Found 12 models, 211 tests, 0 snapshots, 0 analyses, 607 macros, 0 operations, 0 seed files, 8 sources, 0 exposures, 0 metrics
15:38:25
15:38:27  Concurrency: 4 threads (target='dev')
15:38:27
15:38:27  1 of 1 START incremental model DEV.receipts .................................... [RUN]
16:51:02  1 of 1 OK created incremental model DEV.receipts ............................... [SUCCESS 1 in 4354.52s]
16:51:02
16:51:02  Finished running 1 incremental model in 4357.32s.
16:51:02
16:51:02  Completed successfully
16:51:02
16:51:02  Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
```

**Comment by forgxyz** [comment]
Correcting the dockerfile back to `fishtownanalytics`

**Comment by forgxyz** [comment]
Fixed test issue, needed to correctly identify `FLOAT` instead of `NUMBER`, etc.

```
15:39:32  Finished running 211 tests in 66.85s.
15:39:32
15:39:32  Completed with 1 error and 1 warning:
15:39:32
15:39:32  Failure in test sequence_gaps_blocks_block_height (models/core/blocks.yml)
15:39:32    Got 49257 results, configured to fail if >25000
15:39:32
15:39:32    compiled SQL at target/compiled/near/models/core/blocks.yml/sequence_gaps_blocks_block_height.sql
15:39:32
15:39:32  Warning in test tx_gaps_transactions_block_height__txn_hash__tx_count (models/core/transactions.yml)
15:39:32    Got 8 results, configured to warn if between 1 and 25000
15:39:32
15:39:32    compiled SQL at target/compiled/near/models/core/transactions.yml/tx_gaps_transactions_block_height__txn_hash__tx_count.sql
15:39:32
15:39:32  Done. PASS=209 WARN=1 ERROR=1 SKIP=0 TOTAL=211
```

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,5 @@ dbt_docs.sh
 .history/
 **/.DS_Store
 .vscode/ 
+
+.dbt/.user.yml
```
Comment (antonyip) [inline .gitignore:19]: new line.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
-target/*
+target
```
Comment (antonyip) [inline .gitignore:1]: are we keeping something in target now?

**Review by antonyip — APPROVED** [review]
small nits, otherwise ok.

**Pull request merged**